docs/commerce/checkout/extend_checkout.md

```diff
@@ -0,0 +1,48 @@
+---
+description: Extend checkout workflow
+edition: commerce
+---
+
+# Extend checkout workflow
+
+Depending on your needs, the [checkout](checkout.md) process can be customized, as it is highly configurable and extensible. 
+You can create [workflow](workflow.md) definitions under the `framework.workflows` [configuration key](configuration.md#configuration-files). 
+
+To start, use the default workflow that comes with the storefront module as a basis 
+and follow detailed instruction in [Customize checkout](customize_checkout.md).
+
+# Define custom checkout workflow
+
+This example shows how to create custom checkout `new_workflow` and its strategy.
+Defined strategy assumes that the checkout process depends on the currency used in the basket.
+
+# Create custom strategy
+
+Create a PHP definition of the new strategy that allows for workflow manipulation.
+In this example, custom checkout workflow applies when specific currency code ('EUR') is used in the basket. 
```

Comment:
It will be good to add information that defining strategy allows to add conditional step for workflow if needed. Conditional step means that checkout proces will use some provided workflow and go to provided step if condition is true. By default conditional step is set as null, so first step is taken from defined workflow configuration. 
To use conditional step we should pass second argument to constructor in: https://github.com/ibexa/documentation-developer/pull/2111/files#diff-ee2d78df038549c55763550bf73729db81ef4f1da4c10a765d34035b0961d7feR18

```php
return new Workflow(self::IDENTIFIER, 'conditional_step_name');

```